[CORL-2270] Throw better error when removing story with comments

nick-funk opened the pull request:
## What does this PR do?

When removing a story with comments and `includeComments = false/not set`, throws a better error message to indicate our preference that they use `mergeStory` instead.

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

- Create a story
- Create comments
- Call `removeStory` mutation
- See that it throws a descriptive error message
 
## How do we deploy this PR?

No special deploy actions necessary.

alinhle reviewed (changes requested):
Let's talk about why this safeguard is here and whether it makes sense to remove it or instead provide a better error message.

nick-funk committed:
Throw a better error message when removing story

In the case of trying to remove a story that still
has comments with `includeComments` not set, we throw
a descriptive error message telling the user our
preferences that they use `mergeStory` instead.

alinhle commented on src/core/server/services/stories/index.ts:
```diff
@@ -234,8 +234,9 @@ export async function remove(
-    // TODO: (wyattjoh) improve error
-    throw new Error("asset has comments, cannot remove");
+    throw new Error(
+      "Failed to delete story. Attempted to remove a story that still has comments. It is preferable that you merge a story instead of trying to delete it. If you are sure about what you're doing and want to delete this story with its comments. Use the option `includeComments = true`. For more details check the schema suggestions for this mutation input."
```
```suggestion
      "Failed to delete story. Attempted to remove a story that still has comments. It is preferable that you merge a story or update its Story URL instead of trying to delete it. If you are sure about what you're doing and want to delete this story with its comments. Use the option `includeComments = true`. For more details check the schema suggestions for this mutation input."
```

nick-funk committed:
Update src/core/server/services/stories/index.ts

Co-authored-by: Vinh <vinh@vinh.tech>

kodiakhq[bot] committed:
Merge branch 'develop' into feat/CORL-2270-remove-include-comments
